### PR TITLE
Add more action types

### PR DESCRIPTION
**Pull request by drevell**
This PR adds the remaining action types from the design doc: `regex_replace`, `string_replace`, and `go_template`.

### CODE REVIEW

File: templates/model/spec.go [hunk 1]
```diff
@@ -215,10 +225,9 @@ func (s *Step) Validate() error {
 		notZero(s.Pos, s.Desc, "desc"),
 		validateUnlessNil(s.Print),
 		validateUnlessNil(s.Include),
-		// TODO: add more action types:
-		// validateIfNotNil(s.RegexReplace),
-		// validateIfNotNil(s.StringReplace),
-		// validateIfNotNil(s.GoTemplate),
+		validateUnlessNil(s.RegexReplace),
+		validateUnlessNil(s.StringReplace),
+		validateUnlessNil(s.GoTemplate),
```
Comment (yolocs) [inline templates/model/spec.go:230]: Is it worth validating only one of action can be specified? Or it isn't?

File: templates/model/spec.go [hunk 2]
```diff
@@ -288,3 +297,118 @@ func (i *Include) Validate() error {
...
+
+// RegexReplace is an action that replaces a regex match (or a subgroup of it) with a
+// template expression.
+type RegexReplace struct {
+	// Pos is the YAML file location where this object started.
+	Pos *ConfigPos `yaml:"-"`
+
+	Regex    String `yaml:"regex"`
+	Subgroup Int    `yaml:"subgroup"`
+	With     String `yaml:"with"`
```
Comment (yolocs) [inline templates/model/spec.go:309]: Comments (throughout)? Given this file is the source of truth of the spec.

Also, should we add go example later?

File: templates/model/primitive.go [hunk 3]
```diff
@@ -25,6 +25,9 @@ type String = valWithPos[string]
 // Bool represents a boolean field in a model, together with its location in the input file.
 type Bool = valWithPos[bool]
 
+// Bool represents an integer field in a model, together with its location in the input file.
```
Comment (yolocs) [inline templates/model/primitive.go:28]:
```suggestion
// Int represents an integer field in a model, together with its location in the input file.
```

**Review by yolocs — APPROVED** [review]

File: templates/model/spec.go (hunk 1 above)
Comment (drevell) [reply templates/model/spec.go:230]: That's guaranteed by Spec.UnmarshalYAML(), take a look there. We choose a single field to parse into, then call Decode() just once, so only one of them can be set.

File: templates/model/primitive.go (hunk 3 above)
Comment (drevell) [reply templates/model/primitive.go]: Oops, done.

File: templates/model/spec.go (hunk 1 above)
Comment (yolocs) [reply templates/model/spec.go:230]: You mean `Step.UnmarshalYAML()`?  I see it now.

File: templates/model/spec.go (hunk 2 above)
Comment (drevell) [reply templates/model/spec.go:309]: Yeah, we should do that. Let's come back and do documentation once we get something running.

**Pull request merged**